src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFakeImpl.java

```diff
@@ -31,9 +33,25 @@ public void deleteUrlAlias(String email, String alias) {
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Create a list of UrlAlias objects by email.
+   *
+   * @param userEmail gets user email
+   * @return list of UrlAlias objects
+   * @throws edu.kpi.testcourse.storage.UrlRepository.PermissionDenied
+   *         error if emails are not equal
+   */
   @Override
   public List<UrlAlias> getAllAliasesForUser(String userEmail) {
-    // TODO: We should implement it
-    throw new UnsupportedOperationException();
+    List<UrlAlias> aliasesList = new ArrayList<>();
+    for (UrlAlias urlAlias: aliases.values()) {
+      if (urlAlias.email().equals(userEmail)) {
+        aliasesList.add(urlAlias);
+      }
+      else {
+        throw new PermissionDenied();
+      }
```

Comment:
If we want to avoid else statement, we can do it this way (the tests won't fail, I've tested c; )
```suggestion
      if (!urlAlias.email().equals(userEmail)) {
        throw new PermissionDenied();
      }
      aliasesList.add(urlAlias);
      ```

Reply:
Ah, no throws should be there, I'm sorry))
```suggestion
      if (urlAlias.email().equals(userEmail)) {
        aliasesList.add(urlAlias);
      }
```